plugins/GrapesJsBuilderBundle/EventSubscriber/SerializerSubscriber.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MauticPlugin\GrapesJsBuilderBundle\EventSubscriber;
+
+use JMS\Serializer\EventDispatcher\Events;
+use JMS\Serializer\EventDispatcher\EventSubscriberInterface;
+use JMS\Serializer\EventDispatcher\ObjectEvent;
+use JMS\Serializer\JsonSerializationVisitor;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+use Mautic\EmailBundle\Entity\Email;
+use MauticPlugin\GrapesJsBuilderBundle\Integration\Config;
+use MauticPlugin\GrapesJsBuilderBundle\Model\GrapesJsBuilderModel;
+
+class SerializerSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private GrapesJsBuilderModel $grapesJsBuilderModel,
+        private Config $config,
+    ) {
+    }
+
+    /**
+     * @return array<int, array<string, string>>
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            [
+                'event'  => Events::POST_SERIALIZE,
+                'method' => 'addCustomMJML',
+            ],
+        ];
+    }
+
+    public function addCustomMJML(ObjectEvent $event): void
+    {
+        if (!$this->config->isPublished()) {
```

Comment:
I'm not sure why when running the test this is not passing as true and thus all of the response manipulation is not happening for the test. I think there is an issue on how I'm creating the email with grapesJsBuilder. Still looking at how it can be done.

Reply:
Found the way needed to createIntegration.